core: provisioner test failing intermittently

I can get the following failure under load:

```
--- FAIL: TestContext2Apply_provisionerMultiSelfRef (0.03s)
       	context_apply_test.go:2465: bad: []string{"number 0", "number 2", "number 2"}
```

I still have to determine if this is a failure in the test itself, or in the the core logic.

--- a/terraform/eval_apply.go
+++ b/terraform/eval_apply.go
@@ -215,13 +215,13 @@ func (n *EvalApplyProvisioners) apply(ctx EvalContext) error {
 		provisioner := ctx.Provisioner(prov.Type)
 
 		// Interpolate the provisioner config
-		provConfig, err := ctx.Interpolate(prov.RawConfig, n.InterpResource)
+		provConfig, err := ctx.Interpolate(prov.RawConfig.Copy(), n.InterpResource)
 		if err != nil {
 			return err
 		}
 
 		// Interpolate the conn info, since it may contain variables
-		connInfo, err := ctx.Interpolate(prov.ConnInfo, n.InterpResource)
+		connInfo, err := ctx.Interpolate(prov.ConnInfo.Copy(), n.InterpResource)
 		if err != nil {
 			return err
 		}

